### PR TITLE
feat: Add nodejs22.x support

### PR DESCRIPTION
**Pull request by hnnasit**
*Description of changes:*

Add nodejs22.x support

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: nodejs22.x/cw-event/{{cookiecutter.project_name}}/README.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/cw-event/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/hello-img/README.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x`
```
Comment (lucashuy) [inline nodejs22.x/hello-img/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x`
```

File: nodejs22.x/hello-img/{{cookiecutter.project_name}}/README.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the SAM CLI, you need the following tools.
+
+* Docker - [Install Docker community edition](https://hub.docker.com/search/?type=edition&offering=community)
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+
+You may need the following for local testing.
+
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the NPM package management tool.
```
Comment (lucashuy) [inline nodejs22.x/hello-img/{{cookiecutter.project_name}}/README.md:23]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the NPM package management tool.
```

File: nodejs22.x/hello-img/{{cookiecutter.project_name}}/hello-world/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,10 @@
+FROM public.ecr.aws/lambda/nodejs:22
```
Comment (lucashuy) [inline nodejs22.x/hello-img/{{cookiecutter.project_name}}/hello-world/Dockerfile:1]: Did we need the arch here?

File: nodejs22.x/hello-ts-pt/{{cookiecutter.project_name}}/README.md [hunk 5]
```diff
@@ -0,0 +1,182 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+### Deploy the sample application
+
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the SAM CLI, you need the following tools.
+
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the NPM package management tool.
```
Comment (lucashuy) [inline nodejs22.x/hello-ts-pt/{{cookiecutter.project_name}}/README.md:88]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the NPM package management tool.
```

File: nodejs22.x/hello-ts/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x`
```
Comment (lucashuy) [inline nodejs22.x/hello-ts/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x`
```

File: nodejs22.x/hello-ts/{{cookiecutter.project_name}}/README.md [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the SAM CLI, you need the following tools.
+
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the NPM package management tool.
```
Comment (lucashuy) [inline nodejs22.x/hello-ts/{{cookiecutter.project_name}}/README.md:34]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the NPM package management tool.
```

File: nodejs22.x/hello/README.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x`
```
Comment (lucashuy) [inline nodejs22.x/hello/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x`
```

File: nodejs22.x/hello/{{cookiecutter.project_name}}/README.md [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the SAM CLI, you need the following tools.
+
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the NPM package management tool.
```
Comment (lucashuy) [inline nodejs22.x/hello/{{cookiecutter.project_name}}/README.md:34]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the NPM package management tool.
```

File: nodejs22.x/response-streaming/README.md [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x`
```
Comment (lucashuy) [inline nodejs22.x/response-streaming/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x`
```

File: nodejs22.x/response-streaming/{{cookiecutter.project_name}}/README.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/response-streaming/{{cookiecutter.project_name}}/README.md:35]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/s3/README.md [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x --app-template quick-start-s3 --name s3-app`
```
Comment (lucashuy) [inline nodejs22.x/s3/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x --app-template quick-start-s3 --name s3-app`
```

File: nodejs22.x/s3/{{cookiecutter.project_name}}/README.md [hunk 13]
```diff
@@ -0,0 +1,151 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/s3/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/scratch/README.md [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x --app-template quick-start-from-scratch --name sam-app`
```
Comment (lucashuy) [inline nodejs22.x/scratch/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x --app-template quick-start-from-scratch --name sam-app`
```

File: nodejs22.x/scratch/{{cookiecutter.project_name}}/README.md [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/scratch/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/sns/README.md [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x --app-template quick-start-sns --name sns-app`
```
Comment (lucashuy) [inline nodejs22.x/sns/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x --app-template quick-start-sns --name sns-app`
```

File: nodejs22.x/sns/{{cookiecutter.project_name}}/README.md [hunk 17]
```diff
@@ -0,0 +1,146 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/sns/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/sqs/README.md [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x --app-template quick-start-sqs --name sqs-app`
```
Comment (lucashuy) [inline nodejs22.x/sqs/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x --app-template quick-start-sqs --name sqs-app`
```

File: nodejs22.x/sqs/{{cookiecutter.project_name}}/README.md [hunk 19]
```diff
@@ -0,0 +1,141 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/sqs/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

File: nodejs22.x/step-func/README.md [hunk 20]
```diff
@@ -0,0 +1,23 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x`
```
Comment (lucashuy) [inline nodejs22.x/step-func/README.md:16]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x`
```

File: nodejs22.x/step-func/{{cookiecutter.project_name}}/README.md [hunk 21]
```diff
@@ -0,0 +1,137 @@
...
+The AWS Toolkit for VS Code includes full support for state machine visualization, enabling you to visualize your state machine in real time as you build. The AWS Toolkit for VS Code includes a language server for Amazon States Language, which lints your state machine definition to highlight common errors, provides auto-complete support, and code snippets for each state, enabling you to build state machines faster.
+
+## Deploy the sample application
+
+The Serverless Application Model Command Line Interface (SAM CLI) is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda.
+
+To use the SAM CLI, you need the following tools:
+
+* SAM CLI - [Install the SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html)
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the NPM package management tool.
```
Comment (lucashuy) [inline nodejs22.x/step-func/{{cookiecutter.project_name}}/README.md:38]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the NPM package management tool.
```

File: nodejs22.x/web/README.md [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+
+## Requirements
+
+* [AWS SAM CLI](https://github.com/awslabs/aws-sam-cli)
+
+## Usage
+
+Generate a boilerplate template in your current project directory using the following syntax:
+
+* **NodeJS 20**: `sam init --runtime nodejs22.x --app-template quick-start-web --name web-app`
```
Comment (lucashuy) [inline nodejs22.x/web/README.md:13]:
```suggestion
* **NodeJS 22**: `sam init --runtime nodejs22.x --app-template quick-start-web --name web-app`
```

File: nodejs22.x/web/{{cookiecutter.project_name}}/README.md [hunk 23]
```diff
@@ -0,0 +1,160 @@
...
+* [Visual Studio](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/welcome.html)
+
+## Deploy the sample application
+
+The AWS SAM CLI is an extension of the AWS CLI that adds functionality for building and testing Lambda applications. It uses Docker to run your functions in an Amazon Linux environment that matches Lambda. It can also emulate your application's build environment and API.
+
+To use the AWS SAM CLI, you need the following tools:
+
+* AWS SAM CLI - [Install the AWS SAM CLI](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html).
+* Node.js - [Install Node.js 20](https://nodejs.org/en/), including the npm package management tool.
```
Comment (lucashuy) [inline nodejs22.x/web/{{cookiecutter.project_name}}/README.md:36]:
```suggestion
* Node.js - [Install Node.js 22](https://nodejs.org/en/), including the npm package management tool.
```

**Review by lucashuy — APPROVED** [review]

**Review by dkphm — APPROVED** [review]

**Pull request merged**